src/VisualStudio/Core/Def/Options/VisualStudioOptionPersisterProvider.cs

```diff
@@ -46,29 +46,28 @@ internal sealed class VisualStudioOptionPersisterProvider : IOptionPersisterProv
             _featureFlagsService = featureFlagsService;
             _legacyGlobalOptions = legacyGlobalOptions;
             _readFallbacks = readFallbacks.ToImmutableDictionary(item => item.Metadata.ConfigName, item => item);
+            _lazyPersister = new Threading.AsyncLazy<IOptionPersister>(() => CreatePersisterAsync(threadingContext.DisposalToken), threadingContext.JoinableTaskFactory);
```

Comment:
Using DisposalToken here means potentially this could throw a cancellation when the caller didn't ask for one. Hard to say if that'll be bad though, given at that point everything is already more or less shutting down.